x-pack/plugins/ml/server/lib/alerts/register_jobs_monitoring_rule_type.ts

```diff
@@ -151,18 +152,30 @@ export function registerJobsMonitoringRuleType({
       );
       const executionResult = await getTestsResults(options);
 
-      if (executionResult.length > 0) {
+      const unhealthyTests = executionResult.filter(({ isHealthy }) => !isHealthy);
+
+      if (unhealthyTests.length > 0) {
         logger.debug(
-          `"${name}" rule is scheduling actions for tests: ${executionResult
+          `"${name}" rule is scheduling actions for tests: ${unhealthyTests
             .map((v) => v.name)
             .join(', ')}`
         );
 
-        executionResult.forEach(({ name: alertInstanceName, context }) => {
+        unhealthyTests.forEach(({ name: alertInstanceName, context }) => {
           const alertInstance = services.alertFactory.create(alertInstanceName);
           alertInstance.scheduleActions(ANOMALY_DETECTION_JOB_REALTIME_ISSUE, context);
         });
       }
+
+      // Set context for recovered alerts
+      const { getRecoveredAlerts } = services.alertFactory.done();
```

Comment:
When the health rule recovers I don't see any context in the recovered message:

![image](https://user-images.githubusercontent.com/7405507/169516020-44ab1502-8899-4671-acd0-4510767ad37f.png)

I was expecting to see the states of the datafeeds that are now `started`.

Reply:
Fixed in https://github.com/elastic/kibana/pull/132496/commits/67c0a747b8df4a162394a5e0cf2cb6aa1902f140